src/tools/rbd/action/DiskUsage.cc

```diff
@@ -180,9 +180,7 @@ static int do_disk_usage(librbd::RBD &rbd, librados::IoCtx &io_ctx,
           goto out;
         }
 
-        if (snapname != NULL) {
-          total_prov += snap->size;
-        }
+        total_prov += snap->size;
```

Comment:
Not sure if this is correct -- if you don't specify a snapshot, it will also add the image size to the total on line 206 (which would seem like it would double the provisioned total).